atomic_reactor/config.py

```diff
@@ -402,27 +409,34 @@ def registry(self):
                 'insecure': registry.get('insecure', False),
                 'expected_media_types': registry.get('expected_media_types', []),
             }
-            if registry.get('auth'):
-                regdict['secret'] = registry['auth']['cfg_path']
+            if registry.get('auth', False):
+                if (cfg_path := self.registries_cfg_path) is None:
+                    raise ValueError("Missing registries_cfg_path for registry authentication.")
+                regdict['secret'] = cfg_path
             return regdict
 
         raise RuntimeError("Expected V2 registry but none in REACTOR_CONFIG")
 
     @property
-    def yum_proxy(self):
-        return self._get_value(ReactorConfigKeys.YUM_PROXY_KEY, fallback=None)
+    def registries_cfg_path(self) -> Optional[str]:
+        return self._get_value(ReactorConfigKeys.REGISTRIES_CFG_PATH_KEY, fallback=None)
 
-    def _as_source_registry(self, registry):
-        return {
+    def _as_registry(self, registry):
+        info = {
             'uri': RegistryURI(registry['url']),
             'insecure': registry.get('insecure', False),
-            'dockercfg_path': registry.get('auth', {}).get('cfg_path', None)
+            'dockercfg_path': None,
         }
+        if registry.get('auth', False):
+            if (cfg_path := self.registries_cfg_path) is None:
+                raise ValueError("Missing registries_cfg_path for registry authentication.")
+            info['dockercfg_path'] = cfg_path
```

Comment:
What do you think about removing the `auth` param? If a registry is present in the dockercfg file, it uses authentication, otherwise it doesn't. Having another param could be confusing